cmflib/cmfquery.py

```diff
@@ -14,414 +14,670 @@
+        Returns:
+            List of executions for the given stage.
 
+        TODO: Can stages from different pipelines have the same name?. Currently, the first matching stage is used to
+              identify its executions. Also see "get_all_executions_in_stage".
+        """
+        for pipeline in self._get_pipelines():
+            for stage in self._get_stages(pipeline.id):
+                if stage.name == stage_name:
+                    return self.store.get_executions_by_context(stage.id)
+        return []
 
     def get_all_executions_in_stage(self, stage_name: str) -> pd.DataFrame:
-        df = pd.DataFrame()
-        contexts = self.store.get_contexts_by_type("Parent_Context")
-        for ctx in contexts:
-            child_contexts = self.store.get_children_contexts_by_context(ctx.id)
-            for cc in child_contexts:
-                if cc.name == stage_name:
-                    executions = self.store.get_executions_by_context(cc.id)
-                    for exe in executions:
-                        d1 = self._transform_to_dataframe(exe)
-                        # df = df.append(d1, sort=True, ignore_index=True)
-                        df = pd.concat([df, d1], sort=True, ignore_index=True)
+        """Return executions of the given stage as pandas data frame.
 
-        return df
+        Args:
+            stage_name: Stage name.
+        Returns:
+            Data frame with all executions associated with the given stage.
 
-    def get_artifact_df(self, node):
-        d = {"id": node.id, "type": self.store.get_artifact_types_by_id([node.type_id])[0].name, "uri": node.uri,
-             "name": node.name, "create_time_since_epoch": node.create_time_since_epoch,
-             "last_update_time_since_epoch": node.last_update_time_since_epoch}
-        for k, v in node.properties.items():
-            if v.HasField('string_value'):
-                d[k] = v.string_value
-            elif v.HasField('int_value'):
-                d[k] = v.int_value
-            else:
-                d[k] = v.double_value
-        for k, v in node.custom_properties.items():
-            if v.HasField('string_value'):
-                d[k] = v.string_value
-            elif v.HasField('int_value'):
-                d[k] = v.int_value
-            else:
-                d[k] = v.double_value
-        df = pd.DataFrame(d, index=[0, ])
+        TODO: Multiple stages with the same name? This method collects executions from all such stages. Also, see
+              "get_all_exe_in_stage"
+        """
+        df = pd.DataFrame()
+        for pipeline in self._get_pipelines():
+            for stage in self._get_stages(pipeline.id):
+                if stage.name == stage_name:
+                    for execution in self._get_executions(stage.id):
+                        ex_as_df: pd.DataFrame = self._transform_to_dataframe(
+                            execution, {"id": execution.id, "name": execution.name}
```

Comment:
Why this dict as execution.id  which is a value here ? Will this work with the key mapper logic .